spec/index.bs

```diff
@@ -1417,6 +1417,23 @@ applications should generally request the "worst" limits that work for their con
         allowed in {{GPUPipelineLayoutDescriptor/bindGroupLayouts}}
         when creating a {{GPUPipelineLayout}}.
 
+    <tr><td><dfn>maxBindingIndex</dfn>
+        <td>{{GPUSize32}} <td>[=limit class/maximum=] <td>640
+    <tr class=row-continuation><td colspan=4>
+        The maximum value allowed for a
+        {{GPUBindGroupLayoutEntry}}.{{GPUBindGroupLayoutEntry/binding}}
+        when creating a {{GPUBindGroupLayout}}.
+
+        The [=supported limit=] for an [=adapter=] must be &gt;
+        ({{supported limits/maxSampledTexturesPerShaderStage}} +
+        {{supported limits/maxSamplersPerShaderStage}} +
+        {{supported limits/maxStorageBuffersPerShaderStage}} +
+        {{supported limits/maxStorageTexturesPerShaderStage}} +
+        {{supported limits/maxUniformBuffersPerShaderStage}}) for the same [=adapter=].
```

Comment:
hm, I think I got my inequality wrong. for example if all of the per-shader-stage limits were 1, then the sum would be 5. Then maxBindingIndex only has to be &ge; 4, not &gt; 5. Does this make sense:

```suggestion
        The [=supported limit=] for an [=adapter=] must be &ge;
        ({{supported limits/maxSampledTexturesPerShaderStage}} +
        {{supported limits/maxSamplersPerShaderStage}} +
        {{supported limits/maxStorageBuffersPerShaderStage}} +
        {{supported limits/maxStorageTexturesPerShaderStage}} +
        {{supported limits/maxUniformBuffersPerShaderStage}} - 1) for the same [=adapter=].
```

Reply:
TBH this makes me more want to come up with a way to word this so it's actually 640, and not 641. All of the other limits work that way, even `maxInterStageShaderVariables` which limits an index value as well.

Reply:
Ah, yeah. That's annoying. I considered the name `maxBindingIndices` previously, which would (to me) imply that the indices may contain 640 unique values rather than the maximum value being 640. It didn't seem quite as straightforward as `maxBindingIndex` which is why I avoided it, but in light of this issue it's maybe worth considering.

Reply:
shrug, I think `maxBindingIndices` or even `maxBindingsPerBindGroup` is fine. I agree it sounds less strict than it is, but the obvious interpretation isn't _wrong_. Same as `maxInterStageShaderVariables`.

Reply:
Went back and forth on this for a bit and decided that `maxBindingsPerBindGroup` is probably the better name. It matches the pattern of some of the other limits a bit better, and we don't refer to the binding value as a "index" or "indices" anywhere else. Pushed that in the latest commit.